src/lib/bootloader/device_map.rb

```diff
@@ -84,17 +87,22 @@ def propose
     # Grub doesn't support more than 8 devices in device.map
     # @return [Boolean] true if device map was reduced
     def reduce_to_bios_limit
-      if @mapping.size <= BIOS_LIMIT
+      if size <= BIOS_LIMIT
         log.info "device map not need to be reduced"
         return false
       end
 
-      log.info "device map before reduction #{@mapping}"
-      @mapping.select! do |_k, v|
-        v[2..-1].to_i < BIOS_LIMIT
+      grub_devices = @model.grub_devices
+      grub_devices.select! { |d| d.start_with?("hd") }
+      grub_devices.sort_by(&:to_i)
+
+      other_devices = size - grub_devices.size
```

Comment:
What about `other_devices_size`? `grub_devices` is a "collection" and to be consistent, I expected `other_devices` to be another collection.